<?xml version="1.0"?>

<launch>

    <arg name="sim_world" default="singleAgent"/>
    <arg name="sim_robot" default="bot"/>
    <arg name="use_gui" default="false"/>
    <arg name="x_pos" default="0"/>
    <arg name="y_pos" default="0"/>
    <arg name="z_pos" default="0.3"/>
    <arg name="Y_pos" default="1.57"/> 
    <arg name="P_pos" default="0"/>
    <arg name="R_pos" default="0"/>

    <param name="reset_after_collision" type="bool" value="true"/>
    <param name="x_pos" type="double" value="$(arg x_pos)"/>
    <param name="y_pos" type="double" value="$(arg y_pos)"/>
    <param name="z_pos" type="double" value="$(arg z_pos)"/>
    <param name="Y_pos" type="double" value="$(arg Y_pos)"/>
    <param name="randomize_env" type="bool" value="true"/>
    <param name="randomiseRobotSpawning" type="bool" value="true"/>
    

    <!-- gazebo settings -->    

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find simulation_env)worlds/$(arg sim_world).world"/>
        <arg name="use_sim_time" value="false"/>
        <arg name="gui" value="$(arg use_gui)"/>
        <arg name="headless" value="false"/>
        <arg name="recording" value="false"/>
        <arg name="debug" value="false"/>
        
    </include>    

    <!-- robot settings -->


    <!-- coverting xacro to urdf -->
    <arg name="model" default="$(find simulation_env)/urdf/Robot.urdf.xacro"/>
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model) use_nominal_extrinsics:=true add_plug:=true" />



      <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
        <param name="use_gui" value="False"/>
    </node>

    <!-- spawn node  -->   
       <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model"  output="screen" 
          args= "-param robot_description -urdf -model bot -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -R $(arg R_pos) -P $(arg P_pos) -Y $(arg Y_pos)">
        
    </node>

    
    <node pkg="robot_state_publisher" type="robot_state_publisher"  name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="30.0" />
    </node> 
    

                
</launch>